newIDE/app/src/MainFrame/index.js

```diff
@@ -2472,176 +2417,315 @@ const MainFrame = (props: Props) => {
     }
   };
 
-  const createProject = async (
-    i18n: I18n,
-    newProjectSetup: NewProjectSetup
-  ) => {
-    setIsProjectOpening(true);
+  const createProject = React.useCallback(
+    async (i18n: I18n, newProjectSetup: NewProjectSetup) => {
+      setIsProjectOpening(true);
+
+      // 4 cases when creating a project:
+      // - From an example
+      // - From an in-app tutorial
+      // - With the default login enabled
+      // - Empty project
+
+      const selectedInAppTutorialShortHeader =
+        selectedInAppTutorialInfo &&
+        inAppTutorialShortHeaders &&
+        inAppTutorialShortHeaders.find(
+          inAppTutorialShortHeader =>
+            inAppTutorialShortHeader.id === selectedInAppTutorialInfo.tutorialId
+        );
```

Comment:
Should this be done by a a hook or a helper?
Not really the role of `createProject` to fight with the `inAppTutorialShortHeaders` to find some information in it?